mzLib/MzLibUtil/PeriodTolerantFilenameWithoutExtension.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Linq;
+
+namespace MzLibUtil
+{
+    public class PeriodTolerantFilenameWithoutExtension
+    {
+        public static string GetPeriodTolerantFilenameWithoutExtension(string filepathAndOrName)
+        {
+            string windowsSplit = filepathAndOrName.Split('\\').Last();
+            string linuxSplit = filepathAndOrName.Split('/').Last();
```

Comment:
I would use Path.GetFileName to get the name with extension; this will simplify the Linux/Windows path issues. then use LastIndexOf to get the last period. I suppose you could check to see if the extension is .gz, then use LastIndexOf again in that case.